extension/src/fileSystem/index.ts

```diff
@@ -63,6 +64,35 @@ export const findAbsoluteDvcRootPath = async (
   return [absoluteRoot]
 }
 
+// .git inside a submodule is a file with the following content: `gitdir: ../.git/modules/demo`
+const findDotGitDir = (gitRoot: string) => {
+  const dotGitPath = join(gitRoot, gitPath.DOT_GIT)
+
+  const isSubmodule = lstatSync(dotGitPath).isFile()
+  if (isSubmodule) {
+    const dotGitAsFileContent = readFileSync(dotGitPath, 'utf8')
+    const gitDirPrefix = 'gitdir: '
+    const gitDirLine = dotGitAsFileContent
+      .split(/\r?\n/)
+      .find(line => line.indexOf(gitDirPrefix) === 0)
+    return resolve(gitRoot, ...(gitDirLine?.slice(8).split('/') || []))
```

Comment:
[F] I was slightly concerned about this on Windows so wrote a test. It seems to behave as expected on all platforms

Reply:
The reason that I switched from `join` to `resolve` is that I don't want any `..` in the path. It could end giving the watcher the wrong type of `RelativePath`